### PR TITLE
feat: remove observability and cos creation

### PR DESCRIPTION
**Pull request by MatthewLemmond**
### Description
Remove creation of atracker and COS instance+bucket in favor of using the new observability DA for setup

This change is breaking and module will update major version

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [x] Major release (`X.x.x`)

##### Release notes content

Remove the observability and COS modules from the base account module

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: version.tf [hunk 1]
```diff
@@ -2,16 +2,4 @@ terraform {
   required_version = ">= 1.0.0, <1.7.0"
   # If your module requires any terraform providers, uncomment the "required_providers" section below and add all required providers.
   # Each required provider's version should be a flexible range to future proof the module's usage with upcoming minor and patch versions.
-
-  required_providers {
-    ibm = {
-      source  = "IBM-Cloud/ibm"
-      version = ">= 1.53.0, < 2.0.0"
-    }
```
Comment (ocofaigh) [inline version.tf:10]: Is IBM provider not required by the Account settings module?

File: solutions/account-infrastructure-base/main.tf [hunk 2]
```diff
@@ -3,20 +3,7 @@
...
-  cos_bucket_name                     = "${var.prefix}-cos-bucket"
-  trusted_profile_name                = "${var.prefix}-trusted-profile"
-  cos_target_name                     = "${var.prefix}-cos-target"
-  cos_bucket_management_endpoint_type = var.cos_bucket_management_endpoint_type
-  activity_tracker_route_name         = "${var.prefix}-cos-route"
-  kms_key_crn                         = var.kms_key_crn
-  resource_tags                       = var.resource_tags
-  allowed_ip_addresses                = var.allowed_ip_addresses
+  source               = "../.."
+  trusted_profile_name = "${var.prefix}-trusted-profile"
```
Comment (ocofaigh) [inline solutions/account-infrastructure-base/main.tf:7]: lets use this chance to remove the prefix variable and expose a new variable called `trusted_profile_name` (with a default value)

**Review by ocofaigh — CHANGES_REQUESTED** [review]
You will need to remove some variables from solutions/account-infrastructure-base/catalogValidationValues.json.template

**Comment by ocofaigh** [comment]
@MatthewLemmond This will be a major version change. You will need to skip the upgrade test

File: version.tf (hunk 1 above)
Comment (MatthewLemmond) [reply version.tf:10]: pre-commit was flagging an unused provider, but it's not clear why it was saying it's unused so I removed to test it. I can re-add and add an ignore on the unused provider as provider will be needed in the other PR for access groups

Comment (MatthewLemmond) [reply version.tf:10]: re-added and added an ignore for tflint

**Comment by ocofaigh** [comment]
@MatthewLemmond As per internal discussions, lets make Atracker config optional (an opt in feature) instead of removing it completely to cover the case where someone may already have KMS set up in a different account and so could set it up at this early stage

**Comment by MatthewLemmond** [comment]
@ocofaigh given this, I will close this PR and start from the main branch again to make this an opt-in option for the repo/module

**Pull request closed without merging**